internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go

```diff
@@ -223,7 +227,6 @@ func (e *Downloader) downloadFile(ctx context.Context, artifactName, filename, f
 type downloadProgressReporter struct {
 	log         progressLogger
 	sourceURI   string
-	timeout     time.Duration
```

Comment:
This change is unrelated to this PR _per se_.  But I noticed while working on this PR that this field was never being used, so I'm deleting it as part of this PR.